src/server/replica.cc

```diff
@@ -438,33 +442,63 @@ error_code Replica::InitiateDflySync() {
   // Blocked on until all flows got full sync cut.
   fibers_ext::BlockingCounter sync_block{num_df_flows_};
 
+  // Atomically switch to new error handler.
   auto err_handler = [this, sync_block](const auto& ge) mutable {
     sync_block.Cancel();      // Unblock this function.
     DefaultErrorHandler(ge);  // Close sockets to unblock flows.
+    service_.SwitchState(GlobalState::LOADING, GlobalState::ACTIVE);  // Reset global state.
   };
   RETURN_ON_ERR(cntx_.Switch(std::move(err_handler)));
```

Comment:
the error handler here touches  sync_block, which lifetime ends at the end of this function but I dont see a call to cntx_.Switch before this function ends. So if the error handler will run between this function return and the next switch to different error handler it will try to call cancel on the sync_block which is not alive. right?

Reply:
Yes, but its in a shared_ptr